docs/user/debugging.md

```diff
@@ -71,28 +83,143 @@ More information can be found in the [View Logs with Logcat][] help article.
 
-```
+```shell
 adb logcat
 ```
 
 This is the unfiltered output of all log messages.
 You can match for `glean` using grep:
 
-```
+```shell
 adb logcat | grep -i glean
 ```
 
 A simple way to filter for only the application that is being debugged is by using [pidcat][], a wrapper around `adb`, which adds colors and proper filtering by application ID and log level.
 Run it like this to filter for an application:
 
-```
+```shell
 pidcat [applicationId]
 ```
 
 In the above `[applicationId]` is the product's application id as defined in the manifest file and/or build script. For the Glean sample application, this is `org.mozilla.samples.glean` for a release build and `org.mozilla.samples.glean.debug` for a debug build.
 
 [pidcat]: https://github.com/JakeWharton/pidcat
+
+## Debugging the Glean SDK in iOS
+
+For debugging and validation purposes on iOS, Glean makes use of a custom URL scheme which is implemented _within the application_ that is consuming Glean.  Glean provides some convenience functions to facilitate this, but it's up to the consuming application to enable this functionality.  Applications that enable this Glean SDK feature will be able to launch the application from a URL with the Glean debug commands embedded in the URL itself.
+
+### Available commands and query format
+
+There are 3 available commands that you can use with the Glean SDK debug tools
+
+- `logPings`: This is either true or false and will cause pings that are submitted to also be echoed to the device's log
+- `tagPings`: This command will tag outgoing pings with the provided value, in order to identify them in the Glean Debug View. Tags need to be string with upper and lower case letters, numbers and dashes, with a max length of 20 characters.
+- `sendPing`: This command expects a string name of a ping to force immediate collection and submission of.
+
+The structure of the custom URL uses the following format:
+
+`<protocol>://glean?<command 1>=<paramter 1>&<command 2>=<parameter 2> ...`
+
+Where:
+
+- `<protocol>` is the "Url Scheme" that has been added for your app (see Instrumenting the application below), such as `glean-sample-app`.
+- This is followed by `://` and then `glean` which is required for the Glean SDK to recognize the command is meant for it to process.
+- Following standard URL query format, the next character after `glean` is the `?` indicating the beginning of the query.
+- This is followed by one or more queries in the form of `<command>=<parameter>`, where the command is one of the commands listed above, followed by an `=` and then the value or parameter to be used with the command.
+
+There are a few things to consider when creating the custom URL:  
+
+- Invalid commands will log an error and be ignored.
+- Not all commands are requred to be encoded in the URL, you can mix and match the commands that you need.  
+- Special characters should be properly URL encoded and escaped since this needs to represent a valid URL.
```

Comment:
Afaik none of the commands should even accept anything outside of [a-zA-Z0-9-], thus not really needing encoding, right?

Reply:
I think ping names can contain underscores, but that's not usually a problem with URL encoding.  I'll remove this for now since we shouldn't allow characters that require encoding.